lib/internal/cluster/master.js

```diff
@@ -106,25 +115,77 @@ function setupSettingsNT(settings) {
 }
 
 function createWorkerProcess(id, env) {
-  var workerEnv = util._extend({}, process.env);
-  var execArgv = cluster.settings.execArgv.slice();
-  var debugPort = 0;
+  // If master process started with debug, workers shoud use ports with offset
+  // This code parses execArgv from cluster.settings
+  // and augments values of ports in following flags:
+  // --inspect, --inspect-port, --debug, --debug-brk, --debug-port
+
+  const workerEnv = util._extend({}, process.env);
+  const execArgv = cluster.settings.execArgv.slice();
+  const ipv6Regex = /^(\[[\d:]+\])(?::(.+))?$/;
+  let ipv6Match;
+  let debugSocket;
+  let splittedDebugSocket;
+  let debugPort = 0;
+  let debugHost = '';
+  let splittedDebugArg;
+  let resultArg;
```

Comment:
Can you define variables at the place where they're first used instead of upfront?  It's not exactly wrong but since you're using `let` anyway, you might as well profit from lexical scoping.